src/domain/users/user-organizations.repository.ts

```diff
@@ -0,0 +1,305 @@
+import type { FindOptionsWhere, FindOptionsRelations } from 'typeorm';
+import type { IUsersOrganizationsRepository } from '@/domain/users/user-organizations.repository.interface';
+import {
+  UserOrganizationRole,
+  UserOrganizationStatus,
+} from '@/domain/users/entities/user-organization.entity';
+import { getEnumKey } from '@/domain/common/utils/enums';
+import type { Organization } from '@/domain/organizations/entities/organization.entity';
+import type { User } from '@/domain/users/entities/user.entity';
+import type { UserOrganization } from '@/domain/users/entities/user-organization.entity';
+
+@Injectable()
+export class UsersOrganizationsRepository
+  implements IUsersOrganizationsRepository
+{
+  constructor(
+    private readonly postgresDatabaseService: PostgresDatabaseService,
+    @Inject(IUsersRepository)
+    private readonly usersRepository: IUsersRepository,
+    @Inject(IOrganizationsRepository)
+    private readonly organizationsRepository: IOrganizationsRepository,
+    @Inject(IWalletsRepository)
+    private readonly walletsRepository: IWalletsRepository,
+  ) {}
+
+  public async findOneOrFail(
+    where:
+      | Array<FindOptionsWhere<UserOrganization>>
+      | FindOptionsWhere<UserOrganization>,
+    relations?: FindOptionsRelations<UserOrganization>,
+  ): Promise<DbUserOrganization> {
+    const organization = await this.findOne(where, relations);
+
+    if (!organization) {
+      throw new NotFoundException('User organization not found.');
+    }
+
+    return organization;
+  }
+
+  public async findOne(
+    where:
+      | Array<FindOptionsWhere<UserOrganization>>
+      | FindOptionsWhere<UserOrganization>,
+    relations?: FindOptionsRelations<UserOrganization>,
+  ): Promise<DbUserOrganization | null> {
+    const organizationRepository =
+      await this.postgresDatabaseService.getRepository(DbUserOrganization);
+
+    return await organizationRepository.findOne({
+      where,
+      relations,
+    });
+  }
+
+  public async inviteUsers(args: {
+    authPayload: AuthPayload;
+    orgId: Organization['id'];
+    users: Array<{
+      address: `0x${string}`;
+      role: UserOrganizationRole;
+    }>;
+  }): Promise<
+    Array<{
+      userId: User['id'];
+      orgId: Organization['id'];
+      role: keyof typeof UserOrganizationRole;
+      status: keyof typeof UserOrganizationStatus;
+    }>
+  > {
+    const signer = await this.getOrgAndSignerOrFail(args);
+
+    this.assertUserOrgIsActive(signer.userOrg);
+
+    const invitations: Array<{
+      userId: User['id'];
+      orgId: Organization['id'];
+      role: keyof typeof UserOrganizationRole;
+      status: keyof typeof UserOrganizationStatus;
+    }> = [];
+
+    await this.postgresDatabaseService.transaction(async (entityManager) => {
+      for (const userToInvite of args.users) {
+        const invitedUser = await this.walletsRepository.findOneByAddress(
+          userToInvite.address,
+          { user: true },
+        );
```

Comment:
We will move this outside the loop and call `findOrFail` resulting in one query, e.g.

```ts
  const invitedUsers = await this.walletsRepository.findOrFail({
      where: {
        user: {
          id: In(args.users),
        }
      },
      relations: {
        user: true,
      }
  });
```